components/nimbus/src/stateful/enrollment.rs

```diff
@@ -180,6 +180,47 @@ pub fn unenroll_for_pref(
     Ok(events)
 }
 
+pub fn add_previous_state_for_experiment(
+    db: &Database,
+    writer: &mut Writer,
+    experiment_slug: &str,
+    previous_state: PreviousState,
+) -> Result<()> {
+    let enr_store = db.get_store(StoreId::Enrollments);
+
+    if let Ok(Some(existing_enrollment)) =
+        enr_store.get::<ExperimentEnrollment, Writer>(writer, experiment_slug)
+    {
+        // Previous state is only valid on Enrolled experiments
+        let updated_state = &existing_enrollment.on_add_state(previous_state);
+        enr_store.put(writer, experiment_slug, updated_state)?;
+    }
+    Ok(())
+}
+
+pub fn get_previous_state_for_experiment<'r>(
```

Comment:
This is used in one place outside of tests, so we can move the entire thing inline

Reply:
ToDo: Talked on Slack, need to physically inline this.